src/CGRA.cpp

```diff
@@ -170,8 +170,11 @@ CGRA::CGRA(int t_rows, int t_columns, bool t_diagonalVectorization,
     if (t_heterogeneity) {
       for (int r=0; r<t_rows; ++r) {
         for (int c=0; c<t_columns; ++c) {
-          if(r%2==1 and c%2 == 1)
+          if (((r*t_columns)+(c+1)) <= t_pathSupportDim){
```

Comment:
What does this condition indicate? Originally only `[1,1]`, `[1,3]`, `[3,1]`, ..., support heterogeneity. What about now?